packages/core/src/Services/socket-general.js

```diff
@@ -86,6 +88,18 @@ const BinarySocketGeneral = (() => {
         }
     };
 
+    const setSessionDurationLimit = user_limits => {
+        const duration = user_limits?.get_self_exclusion?.session_duration_limit;
```

Comment:
It looks like we are calling `getSelfExclusion` at multiple places. In that case, `setSessionDurationLimit` will be called multiple times.

- If we update `session_duration_limit`. Should we clear the first timeout before creating a new timeout?
- If we update some other value in `self_exclusion` should we still go on to create a new timeout?

Reply:
These are good questions! From my perspective, I think we should clear timeout in case if it's longer than the new `session_duration_limit`, or keep the previous one if it's longer (for one time only). And if we update other values, the new timeout should not be created. I will modify the current logic to accommodate these conditions👍🏻

Reply:
I modified the condition accordingly, so it won't set timeout unless `session_duration_time` is changed. Regarding the first question: it seems like the default behavior of `setTimeout` would be to use the shortest time set, after which it won't try to logout again (e.g. timeout cleared). I think this is already sufficient, as there's no native way to check what is current timeout left, and if we play around with Date object, that would have issues in case if user's system time changes.